Improve List demo description - list editing

czernous-devexpress opened the pull request.

czernous-devexpress committed:
Improve List demos - List Editing (draft v1)

czernous-devexpress committed:
Add selection description (daft v2)

czernous-devexpress committed:
Add deletion description (draft v.3)

czernous-devexpress committed:
Improve List demo - list editing and API

czernous-devexpress committed:
Remove selection description

czernous-devexpress committed:
Delete docUrl

RomanTsukanov commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,26 @@
-The **List** is&nbsp;a&nbsp;UI component that represents a&nbsp;collection of&nbsp;items in&nbsp;a&nbsp;scrollable list. This demo displays different delete modes and shows how you can handle the **selectionChanged** event.
\ No newline at end of file
+The [List](/Documentation/ApiReference/UI_Widgets/dxList/) displays a collection of items as a scrollable list. To create a list, pass an array of items to **dataSource**. 
```
```suggestion
The [List](/Documentation/ApiReference/UI_Widgets/dxList/) displays a collection of items as a scrollable list. To create a list, pass an array of items to the [dataSource](/Documentation/ApiReference/UI_Components/dxList/Configuration/#dataSource) property. 
```

RomanTsukanov commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,26 @@
\ No newline at end of file
+The [List](/Documentation/ApiReference/UI_Widgets/dxList/) displays a collection of items as a scrollable list. To create a list, pass an array of items to **dataSource**. 
+
+This demo illustrates deletion. The UI consists of the **List** and the "Options" section. 
+
+To allow users delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#allowItemDeleting) property to *true*. Use the checkbox in the "Options" section to see how it works.
```
I suggest removing the sentence about the UI because it seems unnecessary if we don't talk about elements in the Options section.

```suggestion
This demo illustrates item deletion. To allow users delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#allowItemDeleting) property to **true**. Click the Allow deleting checkbox under the List to see how this property works.
```

czernous-devexpress committed:
Merge branch '21_1' into list_demo_listEditing_21_1

RomanTsukanov commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,26 @@
+
+To allow users delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#allowItemDeleting) property to *true*. Use the checkbox in the "Options" section to see how it works.
+
+The **List** supports three default and three additional deletion modes. You can see them in action in the "itemDeleteMode" dropdown menu of the "Options" section.
```
Let's give the name of the property as early in the description as possible.

```suggestion
The **List** supports several deletion modes listed below. Use the [itemDeleteMode](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#itemDeleteMode) property to set one of the modes. In this demo, you can select a deletion mode from the itemDeleteMode drop-down menu under the List.
```

RomanTsukanov commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,26 @@
+
+The **List** supports three default and three additional deletion modes. You can see them in action in the "itemDeleteMode" dropdown menu of the "Options" section.
+
+**Default modes**
```
The grouping is unnecessary. We can indicate that one or another mode is default for a platform like this:

`- *"slideItem"* (default for iOS)`

Let's get back to a plain list.

RomanTsukanov commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,26 @@
+**Additional modes**
+
+- *"slideButton"*    
+Users can swipe right to show the **delete** button next to the list item. The content stays in its place.    
```
```suggestion
Swipe left or right to show the Delete button next to the list item. The content stays in its place.    
```

RomanTsukanov commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,26 @@
+- *"slideButton"*    
+Users can swipe right to show the **delete** button next to the list item. The content stays in its place.    
+
+- *"Toggle"*    
```
```suggestion
- *"toggle"*    
```

RomanTsukanov commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,26 @@
+- *"context"*    
+Right-click or touch and hold the list item to call a context menu that contains the **delete** option.    
+
+Use the [itemDeleteMode](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#itemDeleteMode) property to set the deletion mode.
+
+When users delete a list item, it is removed from the "Selected tasks" section. This is done by a function that handles the [onItemDeleted](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#onItemDeleted) event.
```
The `itemDeleteMode` is mentioned before the list now.
Selected tasks are going to be removed as we agreed.

```suggestion
```

czernous-devexpress committed:
Apply suggestions from code review

Co-authored-by: RomanTsukanov <RomanTsukanov@users.noreply.github.com>

czernous-devexpress committed:
Update the description

RomanTsukanov commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
+
+The **List** supports several deletion modes listed below. Use the [itemDeleteMode](/Documentation/ApiReference/UI_Widgets/dxList/Configuration/#itemDeleteMode) property to set one of the modes. In this demo, you can select a deletion mode from the itemDeleteMode drop-down menu under the List.
+    
+- *"static"*    
+Click an **X** button next to a list item to delete it. This is the default mode for desktops. 
```
```suggestion
- *"static"* (default for desktops)    
Click an X button next to a list item to delete it. 
```

RomanTsukanov commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
+Click an **X** button next to a list item to delete it. This is the default mode for desktops. 
+
+- *"slideItem"* (default for iOS)    
+Slide list items left to show the **delete** button. In this case, the list item is moved off the canvas. 
```
```suggestion
Slide list items left to show the Delete button. In this case, the list item is moved off the canvas. 
```

RomanTsukanov commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
+Click a **minus** icon next to a list item to display the **delete** button. 
+
+- *"context"*    
+Right-click or touch and hold the list item to call a context menu that contains the **delete** option.    
```
```suggestion
Right-click or touch and hold the list item to call a context menu that contains the Delete option.    
```

RomanTsukanov commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
+Swipe left or right to show the Delete button next to the list item. The content stays in its place.    
+
+- *"toggle"*    
+Click a **minus** icon next to a list item to display the **delete** button. 
```
```suggestion
Click a **minus** icon next to a list item to display the Delete button. 
```

czernous-devexpress committed:
Apply suggestions from code review

Co-authored-by: RomanTsukanov <RomanTsukanov@users.noreply.github.com>

czernous-devexpress committed:
Apply suggestions from code review

Abadzhev commented on JSDemos/menuMeta.json:
```diff
@@ -5087,7 +5087,6 @@
           {
             "Title": "List Editing and API",
```
Why is the demo titled like this, when you say in the description that it "illustrates item deletion". Can the title be more specific them? 

```suggestion
            "Title": "Item Deletion UI",
```

Abadzhev commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
-The **List** is&nbsp;a&nbsp;UI component that represents a&nbsp;collection of&nbsp;items in&nbsp;a&nbsp;scrollable list. This demo displays different delete modes and shows how you can handle the **selectionChanged** event.
\ No newline at end of file
+The [List](/Documentation/ApiReference/UI_Components/dxList/) displays a collection of items as a scrollable list. To create a list, pass an array of items to the [dataSource](/Documentation/ApiReference/UI_Components/dxList/Configuration/#dataSource) property. 
```
I don't think you need this part here. The demo is not an introduction to lists, it's shows very specific functionality. We don't need to start every demo with a widget definition and basic properties.

czernous-devexpress replied:
It may be better to keep this a text as this demo is the first in the series. This text will serve as an introduction to the List demo series as well as a reminder to users as to how to create a list.

Abadzhev commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
\ No newline at end of file
+The [List](/Documentation/ApiReference/UI_Components/dxList/) displays a collection of items as a scrollable list. To create a list, pass an array of items to the [dataSource](/Documentation/ApiReference/UI_Components/dxList/Configuration/#dataSource) property. 
+
+This demo illustrates item deletion. To allow users delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Components/dxList/Configuration/#allowItemDeleting) property to **true**. Click the Allow deleting checkbox under the List to see how this property works.
```
If you change demo title as I suggest, then you don't really need the first sentence. 

You need to add some formatting to the checkbox option - enclose in quotes or use bold formatting. Otherwise I see that "Allow" is capitalized, but I don't know if "deleting" is still part of the option name. 

Allow deleting is not a great name anyway - see our guidelines on INGs. Allow deletion would be better. 

```suggestion
To allow users to delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Components/dxList/Configuration/#allowItemDeleting) property to **true**. Use the "Allow deletion" checkbox under the List to see how this property works.
```

Abadzhev commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
+
+This demo illustrates item deletion. To allow users delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Components/dxList/Configuration/#allowItemDeleting) property to **true**. Click the Allow deleting checkbox under the List to see how this property works.
+
+The **List** supports several deletion modes listed below. Use the [itemDeleteMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemDeleteMode) property to set one of the modes. In this demo, you can select a deletion mode from the itemDeleteMode drop-down menu under the List.
```
Too many times you need to refer to "modes". Part of the problem is that you start with a format introduction "the component supports". These things don't require much explanation. If you tell people they can select modes, it's understood that the widget supports different modes. So no need in sentence #1. 

Again, I would change "deleting" in the control name to get rid of ING. 

```suggestion
Use the "Deletion UI type" drop-down menu under the List widget to try different deletion behaviors described below. Menu items correspond to [itemDeleteMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemDeleteMode) property values. 
```

Abadzhev commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
+Click an X button next to a list item to delete it. 
+
+- *"toggle"*    
+Click a **minus** icon next to a list item to display the Delete button. 
```
Just to let you know, when I select this mode, I don't see "minus" icons. I know where they should be and I can click that area, but it's empty.

czernous-devexpress replied:
We are aware of this issue and developers are handling it. By the way, the problem goes away on its own when the select functionality is removed from this demo.

Abadzhev commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
+Click a **minus** icon next to a list item to display the Delete button. 
+
+- *"slideButton"*    
+Swipe left or right to show the Delete button next to the list item. The content stays in its place.  
```
```suggestion
Swipe left or right to show the Delete button next to the list item. The content stays in place.  
```

Abadzhev commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
+Swipe left or right to show the Delete button next to the list item. The content stays in its place.  
+
+- *"slideItem"* (default for iOS)    
+Slide list items left to show the Delete button. In this case, the list item is moved off the canvas. 
```
```suggestion
Slide list items left. Item content shifts and the Delete button appears. 
```

czernous-devexpress committed:
Apply suggestions from code review

Co-authored-by: Abadzhev <vladimira@devexpress.com>

czernous-devexpress committed:
Merge branch '21_1' into list_demo_listEditing_21_1

czernous-devexpress committed:
Merge branch '21_1' into list_demo_listEditing_21_1

dirk-pieterse commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
\ No newline at end of file
+The [List](/Documentation/ApiReference/UI_Components/dxList/) displays a collection of items as a scrollable list. To create a list, pass an array of items to the [dataSource](/Documentation/ApiReference/UI_Components/dxList/Configuration/#dataSource) property. 
+
+To allow users to delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Components/dxList/Configuration/#allowItemDeleting) property to **true**. Use the "Allow deletion" checkbox under the List to see how this property works.
```
"to see how this property works" instead of saying what you should see, it is better to state what something does.

czernous-devexpress replied:
Could you provide an example, please?

dirk-pieterse replied:
The "Allow deletion" checkbox under the List does this or that... (what does it do? Saying that it lets you see how something works does not explain what this checkbox does) Does it change something?

dirk-pieterse commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
+
+To allow users to delete items, set the [allowItemDeleting](/Documentation/ApiReference/UI_Components/dxList/Configuration/#allowItemDeleting) property to **true**. Use the "Allow deletion" checkbox under the List to see how this property works.
+
+Use the "Deletion UI type" drop-down menu under the List widget to try different deletion behaviors described below. Menu items correspond to [itemDeleteMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemDeleteMode) property values. 
```
For example,

```suggestion
The "Deletion UI type" drop-down menu under the List widget allows you to switch between the different deletion behaviors described below. Menu items correspond to [itemDeleteMode](/Documentation/ApiReference/UI_Components/dxList/Configuration/#itemDeleteMode) property values. 
```

dirk-pieterse commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
+Click a **minus** icon next to a list item to display the Delete button. 
+
+- *"slideButton"*    
+Swipe left or right to show the Delete button next to the list item. The content stays in place.  
```
"The content stays in place" Do you mean the content does not move? As in, the content is not affected?

czernous-devexpress replied:
Yes. In some cases, list items are moved off-canvas but in this case, they don't move

dirk-pieterse replied:
```suggestion
Swipe left or right to show the Delete button next to the list item. The content of the list item does not move.  
```

czernous-devexpress committed:
Update JSDemos/Demos/List/ListEditingAndAPI/description.md

Co-authored-by: DirkPieterse <dirk.pieterse@devexpress.com>

czernous-devexpress committed:
Update description.md

dirk-pieterse commented on JSDemos/Demos/List/ListEditingAndAPI/description.md:
```diff
@@ -1 +1,23 @@
+Swipe left or right to show the Delete button next to the list item. The content stays in place.  
+
+- *"slideItem"* (default for iOS)    
+Slide list items left. Item content shifts and the Delete button appears. 
```
```suggestion
Slide list items left. The content of the list item shifts to make space for the Delete button. 
```

czernous-devexpress committed:
Apply Dirk's suggestion

changed "...to see how this property works" to "...to toggle this property". The "allow deletion" checkbox toggles the property's value from true to false and vice versa.

czernous-devexpress committed:
Update JSDemos/Demos/List/ListEditingAndAPI/description.md

Co-authored-by: DirkPieterse <dirk.pieterse@devexpress.com>

czernous-devexpress committed:
Update JSDemos/Demos/List/ListEditingAndAPI/description.md

Co-authored-by: DirkPieterse <dirk.pieterse@devexpress.com>
